satpy/etc/readers/hsaf_h5.yaml

```diff
@@ -0,0 +1,29 @@
+reader:
+  name: hsaf_h5
+  short_name: Hydrology SAF
+  long_name: Hydrology SAF products in HDF5 format
+  description: Reader for Hydrology SAF products
+  status: Beta, only h10 currently supported
+  supports_fsspec: false
+  reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader
+  sensors: [hsaf]
+
+file_types:
+  hsafh5:
+    file_reader: !!python/name:satpy.readers.hsaf_h5.HSAFFileHandler
+    file_patterns: ['S-HSAF-h10_{sensing_time:%Y%m%d}_day_merged.H5',
+                    'h10_{sensing_time:%Y%m%d}_day_merged.H5']
+datasets:
+  SC:
+    name: SC
+    msg_name: SC
```

Comment:
I think this is not needed (at least the code doesn't use it). And it has a confusing name :-)